commit 08279d5c9424afd710c90d0b6df95612d2bb5a3f
Author: Mike Marciniszyn <mike.marciniszyn@intel.com>
Date:   Thu Feb 4 10:59:36 2016 -0800

    staging/rdma/hfi1: use new RNR timer
    
    Use the new RNR timer for hfi1.
    
    For qib, this timer doesn't exist, so exploit driver
    callbacks to use the new timer as appropriate.
    
    Reviewed-by: Jubin John <jubin.john@intel.com>
    Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
    Signed-off-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/infiniband/sw/rdmavt/qp.c b/drivers/infiniband/sw/rdmavt/qp.c
index 322de64164f7..439213c37537 100644
--- a/drivers/infiniband/sw/rdmavt/qp.c
+++ b/drivers/infiniband/sw/rdmavt/qp.c
@@ -405,7 +405,6 @@ void rvt_reset_qp(struct rvt_dev_info *rdi, struct rvt_qp *qp,
 
 		/* Stop the send queue and the retry timer */
 		rdi->driver_f.stop_send_queue(qp);
-		del_timer_sync(&qp->s_timer);
 
 		/* Wait for things to stop */
 		rdi->driver_f.quiesce_qp(qp);